### PR TITLE
Weather API Mert Alkan

### PR DESCRIPTION
**Pull request by mertlkn**
Weather functionality works. 

- Urls are set.
- Views.py now has weather function. It is the main code of the functionality.
- Tests are added.

### CODE REVIEW

**Comment by ahmetsenturk** [comment]
Looks okay to me, I think you've implemented an adequate number of test cases. Though, I could not understand how this function works when you have already created two Story instances: 

`
def test_when_openweather_api_fails(self):`

File: practice-app/api/tests.py [hunk 1]
```diff
@@ -1,3 +1,64 @@
...
+
+    def test_when_story_does_not_exist(self):
+        c=Client()
+        resp=c.get("/api/weather/3")
+        self.assertEqual(resp.status_code,404)
+    
+    def test_when_openweather_api_fails(self):
+        c=Client()
+        resp=c.get("/api/weather/2")
+        self.assertEqual(resp.status_code,400)
```
Comment (ahmetsenturk) [inline practice-app/api/tests.py:53]: I could not understand how this function works when you have already created two Story instances.

**Review by ahmetsenturk — COMMENTED** [review]
Looks okay to me, I think you've implemented an adequate number of test cases.

File: practice-app/api/tests.py (hunk 1 above)
Comment (mertlkn) [reply practice-app/api/tests.py:53]: Actually, second Story instance has a problem on its longitude (it being 4100, this is an invalid longitude). Therefore it needs to return status code 400 (call to OpenWeather API did not succeed).

File: practice-app/api/views.py [hunk 2]
```diff
@@ -1,3 +1,69 @@
...
+env = environ.Env(DEBUG=(bool, False))
+
+WEATHER_API_KEY=env('WEATHER_API_KEY')
+
+@csrf_exempt
+def weather(request,story_id):
+
+    if(request.method!='GET'):
+        httpresponse=HttpResponse('Only GET method is available for this API')
+        httpresponse.status_code=400
```
Comment (niyaziulke) [inline practice-app/api/views.py:20]: I think status code 405 could be more appropriate here.

**Review by niyaziulke — COMMENTED** [review]
Seems good to me.

File: practice-app/api/tests.py (hunk 1 above)
Comment (ahmetsenturk) [reply practice-app/api/tests.py:53]: Thanks for the explanation!

**Review by zcanfes — APPROVED** [review]
It looks good to me. Everything is tested and works fine. I think the comments on the weather according to the temperature is a nice addition to this API.

**Review by melihaydogd — APPROVED** [review]
I agree with @niyaziulke about changing status code. Other than that, it looks ok to me.

**Review by melihozcan1 — APPROVED** [review]
API and test functionality seems to be working as planned. Good work.

File: practice-app/api/views.py (hunk 2 above)
Comment (mertlkn) [reply practice-app/api/views.py]: You are right. I edited that part. Thank you!

**Review by niyaziulke — APPROVED** [review]

**Review by omarr09 — APPROVED** [review]
All seems great.

**Comment by mecolak** [comment]
Looks good. Code is understandable and tests are complete.

**Review by mecolak — APPROVED** [review]

**Comment by kkadirkkalkan** [comment]
It is very useful functionality for our project. Code is also nice.

**Review by kkadirkkalkan — APPROVED** [review]

**Pull request merged**